### ISSUE TITLE
Size of code changes post-review weight should compare with the previously confirmed version

### ISSUE THREAD

**Issue by erosman** [mozilla/addons#616]
**Edited by @diox:** Size of code change is calculated by comparing the version we're approving to the `current_version`. That's not good: we need to compare it to the latest version to have been confirmed by a human, otherwise an add-on can ship a massive update and then quickly ship a second version with almost no changes to get a reduced weight - since we only care about the most recent version.

Original issue as filed by @erosman below:
> It seems the Content Approval is drastically reducing the weighting while it shouldn't.
> The code has not been checked and the code should be the primary focus of the weighting.
> The following are some examples. While the code in the following are not a problem, dropping the weighting from **200** to **0** (in some cases, but usually bellow 20 margin) just because of 'Content approved', can lead to all sorts of issues getting overlooked since the code has not been checked by a reviewer.
Frankly, IMHO, "Content rejection' could increase the weighting but 'Content approved' shouldn't affect the weighting.

**Comment by diox** [comment]
That... should not be the case. At all. Content Approval should have no influence over a version weight. I'll investigate.

**Comment by diox** [comment]
Yeah, I think what's happening here is that the weight for the size of code changes is miscalculated. When I coded it I naively took the add-on's `current_version` as the base, and compare it to the newest version being analyzed. That's not good: we need to compare it to the latest version to have been confirmed by a human, I think.

**Comment by AlexandraMoga** [comment]
Verified this on AMO-dev with FF57, (Win10)

I've submitted the following add-on with 3 versions: https://reviewers.addons-dev.allizom.org/en-US/reviewers/review/code_x
- v1 - code size =>164 bytes - assigned weight = 0 - auto-approval confirmed
- v2 - code size => 9575 bytes - assigned weight = 11* - auto-approved (not confirmed)
- v3 - code size => 12813 bytes - assigned weight = 12

Conclusions:
- @diox for v2 I couldn't figure it out exactly where the extra 10 weight is coming from. I didn't receive any warnings during add-on validation
- application of weight according to code size seams to be calculated correctly. v3 receives 2 weight for code changes, which is the expected weight if we compare it with the code size in v1. If the comparison was made with v2, then there wouldn't have been a weight increase for v3,
